protocol/src/format.rs

```diff
@@ -369,20 +369,20 @@ pub enum ComponentType {
 }
 
 impl ComponentType {
-    pub fn new(val: &str) -> Self {
-        Self::Text {
-            text: val.to_owned(),
-            modifier: Modifier::default(),
-        }
-    }
-
-    pub fn new_with_color(val: &str, color: Color) -> Self {
-        Self::Text {
-            text: val.to_owned(),
-            modifier: Modifier {
-                color,
-                ..Default::default()
-            },
+    pub fn new(val: &str, color: Option<Color>) -> Self {
+        if let Some(color) = color {
```

Comment:
This can be simplefied, as much of the code is duplicated here, as it just diverges in the modifier field value

Reply:
You are thinking like this ? 

```rust
    pub fn new(val: &str, color: Option<Color>) -> Self {
        Self::Text {
            text: val.to_string(),
            modifier: match color {
                Some(color) => Modifier { color, .. },
                None => Modifier::default(),
            },
        }
    }
```

Reply:
exactly!